libs/app-utils/main/src/lib/app-http/app-http.service.ts

```diff
@@ -3,30 +3,53 @@
  * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
  */
 
-import axios from "axios"
-import { AppHttpRequestConfig, AppHttpResponse } from "app-utils/models"
+import axios, { AxiosError } from "axios"
+import {
+  AppError,
+  AppErrorType,
+  AppHttpFailedResult,
+  AppHttpRequestConfig,
+  AppHttpResult,
+} from "app-utils/models"
 
-// TODO: Implement AppError (or ObjectResult pattern)
-const isError = (error: unknown): error is Error => {
-  return error !== null && typeof error === "object" && "type" in error
-}
-
-const mapToError = (error: unknown): Error => {
-  if (isError(error)) {
-    return error
-  }
-  return new Error("Unknown error occurred")
+// TODO: Candidate for a utility function
+const getErrorMessage = (error: unknown): string => {
+  if (typeof error === "string") return error
+  if (error instanceof Error) return error.message
+  return String(error)
 }
 
 export class AppHttpService {
-  async request<T = unknown>(
+  static async request<Data = unknown>(
     config: AppHttpRequestConfig
-  ): Promise<AppHttpResponse<T> | Error> {
+  ): Promise<AppHttpResult<Data>> {
     try {
-      const { data, status } = await axios.request<T>(config)
-      return { data, status }
+      const { data, status } = await axios.request<Data>(config)
+      return { ok: true, data, status }
     } catch (error) {
-      return mapToError(error)
+      return AppHttpService.mapAppHttpFailedResult<Data>(error)
+    }
+  }
+  private static isAxiosError(error: unknown): error is AxiosError {
+    return (
+      typeof error === "object" &&
+      error !== null &&
+      "isAxiosError" in error &&
+      (error as any).isAxiosError === true
+    );
+  }
+
+  private static mapAppHttpFailedResult<ErrorData = unknown>(
+    error: unknown
+  ): AppHttpFailedResult<AppErrorType, ErrorData> {
+    if (this.isAxiosError(error)) {
+      return {
+        ok: false,
+        error: new AppError(error.message),
+        data: error.response?.data as ErrorData,
+        status: error.response?.status,
+      };
     }
+    return { ok: false, error: new AppError(getErrorMessage(error)) };
```

Comment:
`getErrorMessage` returns a message from Error, but the first argument of AppError is `type`